src/base-element.js

```diff
@@ -950,4 +950,8 @@ export class BaseElement {
    * @public
    */
   onLayoutMeasure() {}
+
+  user() {
```

Comment:
I get @lannka suggestion on using `user(this).error()`. But then I thought we are going to use `user(ampdoc).error()` in future. Then we have `this.user().error()` `user(element).error()` `user(ampdoc).error()` which is confusing to me. 
Do you think it's better to let `user().error` to accept `{Element|AmpElement|Ampdoc} instead, and if it is AmpElement, we can se `user(AmpElement.element)`. Or I think it makes sense to only support Ampdoc. 
cc @lannka